### ISSUE TITLE
[a11y]: Date picker - unable to use keybord after opening calendar

### ISSUE THREAD

**Issue by marcinmotNB** [carbon-design-system/carbon#9815]
### Package

carbon-components

### Browser

Chrome, Firefox

### Operating System

Windows

### Package version

10.42.0

### Automated testing tool and ruleset

manual

### Description

When user clicks on calendar icon in date picker it;s impossible to pick date with arrows on keyboard.
When calendar is opened by clicking on input field keyboard select works fine.

Opened by icon clicked:
![image](https://user-images.githubusercontent.com/43094706/136212968-0ddd65db-99b4-4ff9-a360-44be37f8b0d3.png)

Opened by input cliecked:
![image](https://user-images.githubusercontent.com/43094706/136213166-03ab7b0f-7f9a-4e41-8fa5-0cf3b3c1b083.png)

### CodeSandbox example

https://codesandbox.io/s/withered-rain-q2olg

### Steps to reproduce

Click on calendar icon in date picker.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
